src/common/pool_map.c

```diff
@@ -451,23 +456,29 @@ pool_buf_unpack(struct pool_buf *buf)
 static int
 pool_buf_parse(struct pool_buf *buf, struct pool_domain **tree_pp)
 {
-	struct pool_domain *tree;
-	struct pool_domain *domain;
-	struct pool_domain *parent;
-	struct pool_target *targets;
-	pool_comp_type_t    type;
-	int		    size;
-	int		    i;
-	int		    rc = 0;
+	struct pool_domain	*tree;
+	struct pool_domain	*domain;
+	struct pool_domain	*parent;
+	struct pool_target	*targets;
+	struct pool_component	*comp;
+	pool_comp_type_t	 type;
+	int			 size;
+	int			 i, nr;
+	int			 rc = 0;
 
 	if (buf->pb_target_nr == 0 || buf->pb_node_nr == 0 ||
-	    buf->pb_domain_nr + buf->pb_node_nr + buf->pb_target_nr !=
-								buf->pb_nr) {
+	    buf->pb_domain_nr + buf->pb_node_nr + buf->pb_target_nr != buf->pb_nr) {
 		D_DEBUG(DB_MGMT, "Invalid number of components: %d/%d/%d/%d\n",
 			buf->pb_nr, buf->pb_domain_nr, buf->pb_node_nr,
 			buf->pb_target_nr);
 		return -DER_INVAL;
 	}
+	if (buf->pb_domain_nr != 0 && buf->pb_comps[0].co_type != PO_COMP_TP_GRP &&
+	    buf->pb_comps[0].co_type != PO_COMP_TP_NODE) {
+		D_DEBUG(DB_MGMT, "Invalid co_type %s[%d] for first domain.\n",
```

Comment:
ok, just because original code at above L471 use debug loglevel so I also use debug here. Can change it here later if need to refresh. But here is for invalid arg check for very cornel case.